File: JuegoAhorcado/JuegoAhorcadoBlazorServer/Services/PlayerDeserializer.cs
using JuegoAhorcadoBlazorServer.Modelo;

namespace JuegoAhorcadoBlazorServer.Services
{
    public class PlayerDeserializer
    {
        public Player Deserialize(string text)
        {
            if (string.IsNullOrWhiteSpace(text))
                throw new ArgumentException("El texto del Player está vacío.");

            string[] values = text.Split('|');

            if (values.Length != 7)
                throw new Exception("Formato inválido para Player. Se esperaban 7 valores.");

            return new Player
            {
                Id = int.Parse(values[0]),
                Name = values[1],
                Password = values[2],
                Role = new Role
                {
                    Id = int.Parse(values[3]),
                    Name = values[4]
                },
                Rounds = int.Parse(values[5]),
                Points = int.Parse(values[6])
            };
        }
    }
}

File: AhorcadoJuegoBackend/src/juegoUtils/PlayerDeserializer.java
package juegoUtils;

import models.Player;
import services.Deserializer;

public class PlayerDeserializer implements Deserializer {

	@Override
    public <T> T deserealize(String t, Class<T> c) {
        try {
            String[] values = t.split("\\%");

            int id = Integer.parseInt(values[0]);
            String name = values[1];
            String pass = values[2];
            String role = values[3];
            int rounds = Integer.parseInt(values[4]);
            int points = Integer.parseInt(values[5]);

            Player player = new Player(id, name, pass, role, rounds, points);

            return (T) player;

        } catch (Exception e) {
            throw new RuntimeException(e.getMessage());
        }
    }

}


// auth/register/player=0|pepe|123|player|0|0